### PR TITLE
DOC: pin sphinx to 1.8.5

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/26058

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Context for doing this would be the styling changes introduced in sphinx 2, for which we should first update our custom css (described in #26058)

**Comment by jorisvandenbossche** [comment]
cc @datapythonista @TomAugspurger 

If we would release right now, I would build the docs with sphinx 1.8.5 for the release. 
Given that, the question is then: do we want to use 1.8.5 in the travis build as well, to make sure everything is working with that release (to match our release process), or do we rather already want to build with the newest possible sphinx to also catch possible new errors with newer sphinx? (both have value IMO)

**Comment by TomAugspurger** [comment]
I think pin to 1.8.5 for now.

**Comment by jreback** [comment]
Agreed pinning is fine (i think we have pinned sphinx in the past frequently)

**Review by datapythonista — COMMENTED** [review]
+1 on pinning

I'd probably migrate to sphinx 2 when we change the theme. And surely after we fail the CI on doc warnings.

But can you also pin the version in `environment.yml` please, so we build the docs locally with the same version as travis.

**Review by datapythonista — COMMENTED** [review]
and may be update the comment on pinning to why we're using 1.8.5, and duplicate it in `environment.yml` too (I'll propose getting rid of this deps file again soon).

**Comment by jorisvandenbossche** [comment]
Updated

**Review by datapythonista — APPROVED** [review]
lgtm, thanks

**Comment by phamvantuong** [comment]
Ok travic job very good

Vào 12 Th6, 2019 13:47, Marc Garcia <notifications@github.com> đã viết:

@datapythonista commented on this pull request.

+1 on pinning

I'd probably migrate to sphinx 2 when we change the theme. And surely after we fail the CI on doc warnings.

But can you also pin the version in environment.yml please, so we build the docs locally with the same version as travis.

File: ci/deps/travis-36-doc.yaml [hunk 1]
```diff
@@ -33,8 +33,8 @@ dependencies:
   - pytz
   - scipy
   - seaborn
-  # recursion error with sphinx 2.1.0. https://github.com/pandas-dev/pandas/issues/26723
-  - sphinx==2.0.1
+  # some styling is broken with sphinx >= 2 (https://github.com/pandas-dev/pandas/issues/26058)
+  - sphinx==1.8.5
```
Comment (datapythonista) [inline ci/deps/travis-36-doc.yaml]: In conda files the version is specified with just one equal (`==` is for pip, a bit confusing)

You'll also have to call `./scripts/generate_pip_deps_from_conda.py` to update `requirements_dev.txt`, which is synched from `environment.yml`

Comment (jorisvandenbossche) [reply ci/deps/travis-36-doc.yaml]: Ah, thanks!

Comment (datapythonista) [reply ci/deps/travis-36-doc.yaml]: you also need to remove one equal here, other than that looks good

Comment (jorisvandenbossche) [reply ci/deps/travis-36-doc.yaml]: Just wondering: does it actually matter? (it was with double equals before as well)

Comment (datapythonista) [reply ci/deps/travis-36-doc.yaml]: not sure, but we've got a function to convert them from one to two when synching, if we can have two may be better to leave two everywhere and simplify the synching

**Comment by datapythonista** [comment]
thanks @jorisvandenbossche

**Pull request merged**